### PR TITLE
techdebt: [PL-31832]: Fixes 2nd batch of cycles in 930-delegate-tasks

### PR DESCRIPTION
**Pull request by Raghav-Murali**
## Describe your changes

This PR fixes 2nd batch of upto 35 cycles in 930-delegate-tasks. It also adds granular build targets for some modules that are compilable.

Most of the refactoring have been done without moving the packages. However, the below 5 classes involved moving packages -

1. 930-delegate-tasks/src/main/java/software/wings/delegatetasks/helm/ArtifactoryHelmRepositoryService.java
io.harness.perpetualtask.manifest; —> software.wings.delegatetasks.helm;

2. 930-delegate-tasks/src/main/java/software/wings/delegatetasks/helm/HelmRepoConfigValidationTask.java
software.wings.delegatetasks; -> software.wings.delegatetasks.helm;

3. 930-delegate-tasks/src/main/java/software/wings/helpers/ext/bamboo/BambooCollectionTask.java
software.wings.delegatetasks.collect.artifacts; -> software.wings.helpers.ext.bamboo;

4. 930-delegate-tasks/src/main/java/software/wings/helpers/ext/helm/request/ArtifactoryHelmTaskHelper.java
software.wings.delegatetasks.helm; -> software.wings.helpers.ext.helm.request;

5. 930-delegate-tasks/src/main/java/software/wings/service/impl/instana/InstanaDataCollector.java
software.wings.delegatetasks.cv; -> software.wings.service.impl.instana;

Tested the end-to-end k8s deployment workflow in PR env.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45640)

### CODE REVIEW

**Comment by Raghav-Murali** [comment]
trigger messagecheck

**Comment by Raghav-Murali** [comment]
trigger messagecheck

**Review by mdmilic — CHANGES_REQUESTED** [review]
> Reviewed 45 of 154 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45640)* status: 154 file(s) to review. 1 change requests. 0 of 2 approvals obtained. Waiting on team(s): del-team. 

___
*[`930-delegate-tasks/src/main/java/io/harness/beans/BUILD.bazel` line 33 at r1](https://reviewable.io/reviews/harness/harness-core/45640#-NRi3QgNBzxI0atEirNi:-NRi3QgO-MxylOQLqswO:b-pwy6ak) ([raw file](https://github.com/harness/harness-core/blob/b64ebd76a5eb71fa4ccb99f1631c3f4b3d291926/930-delegate-tasks/src/main/java/io/harness/beans/BUILD.bazel#L33)):*
> ```Starlark
>         "@maven//:com_google_guava_guava",
>         "@maven//:dev_morphia_morphia_core",
>         "@maven//:io_harness_ng_license_beans",
> ```

This one sometimes comes up but it's wrong, you should remove it. 
Probably morphia shouldn't be needed either, we want delegate morphia free
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/exceptionhandler/core/BUILD.bazel` line 19 at r1](https://reviewable.io/reviews/harness/harness-core/45640#-NRi3ddS0zMZ7JRLoRon:-NRi3ddS0zMZ7JRLoRoo:bb7od0r) ([raw file](https://github.com/harness/harness-core/blob/b64ebd76a5eb71fa4ccb99f1631c3f4b3d291926/930-delegate-tasks/src/main/java/io/harness/delegate/exceptionhandler/core/BUILD.bazel#L19)):*
> ```Starlark
>         "//980-commons/src/main/java/io/harness/exception/exceptionmanager:module",
>         "@maven//:com_google_inject_guice",
>         "@maven//:io_harness_ng_license_beans",
> ```

same
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/common/BUILD.bazel` line 29 at r1](https://reviewable.io/reviews/harness/harness-core/45640#-NRi3kU540AvHHW7vR0x:-NRi3kU540AvHHW7vR0y:bb7eybp) ([raw file](https://github.com/harness/harness-core/blob/b64ebd76a5eb71fa4ccb99f1631c3f4b3d291926/930-delegate-tasks/src/main/java/io/harness/delegate/task/common/BUILD.bazel#L29)):*
> ```Starlark
>         "//980-commons/src/main/java/io/harness/secret:module",
>         "@maven//:com_google_inject_guice",
>         "@maven//:io_harness_ng_license_beans",
> ```

dito
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/shell/ssh/command/BUILD.bazel` line 16 at r1](https://reviewable.io/reviews/harness/harness-core/45640#-NRi3sCcCcYCOBqOG66p:-NRi3sCcCcYCOBqOG66q:brq253x) ([raw file](https://github.com/harness/harness-core/blob/b64ebd76a5eb71fa4ccb99f1631c3f4b3d291926/930-delegate-tasks/src/main/java/io/harness/delegate/task/shell/ssh/command/BUILD.bazel#L16)):*
> ```Starlark
>         "//950-delegate-tasks-beans:module",
>         "//970-api-services-beans/src/main/java/io/harness/shell:module",
>         "@maven//:io_harness_ng_license_beans",
> ```

make sure no bazel files have this dep
___
*[`930-delegate-tasks/src/main/java/software/wings/beans/apm/BUILD.bazel` line 29 at r1](https://reviewable.io/reviews/harness/harness-core/45640#-NRi4CMt0mFFLQkl8xM5:-NRi4CMt0mFFLQkl8xM6:b-1yn6n8) ([raw file](https://github.com/harness/harness-core/blob/b64ebd76a5eb71fa4ccb99f1631c3f4b3d291926/930-delegate-tasks/src/main/java/software/wings/beans/apm/BUILD.bazel#L29)):*
> ```Starlark
>         "@maven//:com_fasterxml_jackson_core_jackson_annotations",
>         "@maven//:com_github_reinert_jjschema",
>         "@maven//:dev_morphia_morphia_core",
> ```

morphia probably shouldn't be in 930 anywhere either.

**Review by Raghav-Murali — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45640)* status: 154 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/beans/BUILD.bazel` line 33 at r1](https://reviewable.io/reviews/harness/harness-core/45640#-NRi3QgNBzxI0atEirNi:-NRjTFkJBMFBGwEzrp-w:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/b64ebd76a5eb71fa4ccb99f1631c3f4b3d291926/930-delegate-tasks/src/main/java/io/harness/beans/BUILD.bazel#L33)):*
<details><summary><i>Previously, mdmilic (Marko Milic) wrote…</i></summary><blockquote>

This one sometimes comes up but it's wrong, you should remove it. 
Probably morphia shouldn't be needed either, we want delegate morphia free
</blockquote></details>

Done.



_Code quote:_
```Starlark
@maven//:dev_morphia_morphia_core
```
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/exceptionhandler/core/BUILD.bazel` line 19 at r1](https://reviewable.io/reviews/harness/harness-core/45640#-NRi3ddS0zMZ7JRLoRon:-NRjTKl6BuY5uTEpzgU6:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/b64ebd76a5eb71fa4ccb99f1631c3f4b3d291926/930-delegate-tasks/src/main/java/io/harness/delegate/exceptionhandler/core/BUILD.bazel#L19)):*
<details><summary><i>Previously, mdmilic (Marko Milic) wrote…</i></summary><blockquote>

same
</blockquote></details>

Done.



_Code quote (from 930-delegate-tasks/src/main/java/io/harness/beans/BUILD.bazel):_
```Starlark
dev_morphia_morphia_core
```
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/common/BUILD.bazel` line 29 at r1](https://reviewable.io/reviews/harness/harness-core/45640#-NRi3kU540AvHHW7vR0x:-NRjTLYN95mWxb77jWER:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/b64ebd76a5eb71fa4ccb99f1631c3f4b3d291926/930-delegate-tasks/src/main/java/io/harness/delegate/task/common/BUILD.bazel#L29)):*
<details><summary><i>Previously, mdmilic (Marko Milic) wrote…</i></summary><blockquote>

dito
</blockquote></details>

Done.
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/shell/ssh/command/BUILD.bazel` line 16 at r1](https://reviewable.io/reviews/harness/harness-core/45640#-NRi3sCcCcYCOBqOG66p:-NRjTMtk5k4u52fzDfdR:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/b64ebd76a5eb71fa4ccb99f1631c3f4b3d291926/930-delegate-tasks/src/main/java/io/harness/delegate/task/shell/ssh/command/BUILD.bazel#L16)):*
<details><summary><i>Previously, mdmilic (Marko Milic) wrote…</i></summary><blockquote>

make sure no bazel files have this dep
</blockquote></details>

Done
___
*[`930-delegate-tasks/src/main/java/software/wings/beans/apm/BUILD.bazel` line 29 at r1](https://reviewable.io/reviews/harness/harness-core/45640#-NRi4CMt0mFFLQkl8xM5:-NRjTOLO5pHeY2rR0fWQ:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/b64ebd76a5eb71fa4ccb99f1631c3f4b3d291926/930-delegate-tasks/src/main/java/software/wings/beans/apm/BUILD.bazel#L29)):*
<details><summary><i>Previously, mdmilic (Marko Milic) wrote…</i></summary><blockquote>

morphia probably shouldn't be in 930 anywhere either.
</blockquote></details>

Done

**Review by mdmilic — DISMISSED** [review]
> Reviewed 93 of 154 files at r1, 48 of 48 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45640)* status: 155 file(s) to review. 1 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/software/wings/delegatetasks/gcb/BUILD.bazel` line 31 at r2](https://reviewable.io/reviews/harness/harness-core/45640#-NRjeaysCajQSY2GMZpa:-NRjeaysCajQSY2GMZpb:b-qjs275) ([raw file](https://github.com/harness/harness-core/blob/8c3533e9f3062bfca74f0b8830d7d5f6cc91e5b9/930-delegate-tasks/src/main/java/software/wings/delegatetasks/gcb/BUILD.bazel#L31)):*
> ```Starlark
>         "//980-commons/src/main/java/io/harness/threading:module",
>         "@maven//:com_google_inject_guice",
>         "@maven//:com_intellij_annotations",
> ```

Can you just double check where is this one coming from. Sometimes it can be picked for some common things like @Slf4j IIRC instead of depending on lombok. Sometimes developers just wrongly use intellijs @Nonnull, but don't bother fixing that if it's the case
___
*[`930-delegate-tasks/src/main/java/software/wings/service/intfc/security/BUILD.bazel` line 25 at r2](https://reviewable.io/reviews/harness/harness-core/45640#-NRjfoiPCltFgw7ORCZK:-NRjfoiQ8e6iOhoyVlD7:b-ksz83q) ([raw file](https://github.com/harness/harness-core/blob/8c3533e9f3062bfca74f0b8830d7d5f6cc91e5b9/930-delegate-tasks/src/main/java/software/wings/service/intfc/security/BUILD.bazel#L25)):*
> ```Starlark
>         "//950-delegate-tasks-beans:module",
>         "//957-cg-beans:module",
>         "//960-persistence:module",
> ```

hmm, can you double check the usage of 960-persistence why it's coming up. I thought we removed it from delegate.

**Comment by Raghav-Murali** [comment]
trigger feature-build

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/common/BUILD.bazel [hunk 1]
```diff
@@ -16,16 +16,15 @@ java_library(
         "//930-delegate-tasks/src/main/java/io/harness/delegate/task/tasklogging:module",
         "//950-delegate-tasks-beans:module",
         "//955-delegate-beans:module",
-        "//957-cg-beans/src/main/java/io/harness/delegate/beans:module",
-        "//970-api-services-beans:module",
+        "//957-cg-beans:module",
```
Comment (gn-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/common/BUILD.bazel:19]: why this change?

File: 930-delegate-tasks/src/main/java/software/wings/helpers/ext/bamboo/BambooCollectionTask.java [hunk 2]
```diff
@@ -5,7 +5,7 @@
  * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
  */
 
-package software.wings.delegatetasks.collect.artifacts;
+package software.wings.helpers.ext.bamboo;
```
Comment (gn-harness) [inline 930-delegate-tasks/src/main/java/software/wings/helpers/ext/bamboo/BambooCollectionTask.java:8]: all these package name changes will lead to Kryo exceptions, right?

**Comment by mdmilic** [comment]
trigger feature-build

**Review by Raghav-Murali — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45640)* status: 155 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/common/BUILD.bazel` line 19 at r2](https://reviewable.io/reviews/harness/harness-core/45640#-NRREFJYh3vlkIX6EhK8-r2-19:-NRo8pBF1aZ2E1H6_-_g:bnjadsj) ([raw file](https://github.com/harness/harness-core/blob/8c3533e9f3062bfca74f0b8830d7d5f6cc91e5b9/930-delegate-tasks/src/main/java/io/harness/delegate/task/common/BUILD.bazel#L19)):*
<details><summary><i>Previously, gn-harness (Gaurav Nanda) wrote…</i></summary><blockquote>

why this change?
</blockquote></details>

Ah, actually I had a local BuildCleaner change to modify all 957-cg-beans to its parent module since not all 957-cg-beans modules had a build target. That has led to this, I will check it and update.
___
*[`930-delegate-tasks/src/main/java/software/wings/delegatetasks/gcb/BUILD.bazel` line 31 at r2](https://reviewable.io/reviews/harness/harness-core/45640#-NRjeaysCajQSY2GMZpa:-NRo8ZbO2M__dR7UOU7e:b-vx9ol6) ([raw file](https://github.com/harness/harness-core/blob/8c3533e9f3062bfca74f0b8830d7d5f6cc91e5b9/930-delegate-tasks/src/main/java/software/wings/delegatetasks/gcb/BUILD.bazel#L31)):*
<details><summary><i>Previously, mdmilic (Marko Milic) wrote…</i></summary><blockquote>

Can you just double check where is this one coming from. Sometimes it can be picked for some common things like @Slf4j IIRC instead of depending on lombok. Sometimes developers just wrongly use intellijs @Nonnull, but don't bother fixing that if it's the case
</blockquote></details>

Yes, the below 2 files use `org.jetbrains.annotations` thus the dependency -
1. AzureAsyncTaskHelper.java
2. GcbDelegateResponse.java
___
*[`930-delegate-tasks/src/main/java/software/wings/service/intfc/security/BUILD.bazel` line 25 at r2](https://reviewable.io/reviews/harness/harness-core/45640#-NRjfoiPCltFgw7ORCZK:-NRo7pfB4l1oHekesaVU:b-3acu7v) ([raw file](https://github.com/harness/harness-core/blob/8c3533e9f3062bfca74f0b8830d7d5f6cc91e5b9/930-delegate-tasks/src/main/java/software/wings/service/intfc/security/BUILD.bazel#L25)):*
<details><summary><i>Previously, mdmilic (Marko Milic) wrote…</i></summary><blockquote>

hmm, can you double check the usage of 960-persistence why it's coming up. I thought we removed it from delegate. 
</blockquote></details>

That's a good catch. Need to look into the BuildCleaner a bit on why the 960-persistence module got added. There is one dependency on it though -

In `930-delegate-tasks/src/main/java/io/harness/secrets/setupusage/BUILD.bazel` has below dependency -
"//960-persistence/src/main/java/io/harness/persistence/api:module"
___
*[`930-delegate-tasks/src/main/java/software/wings/helpers/ext/bamboo/BambooCollectionTask.java` line 8 at r2](https://reviewable.io/reviews/harness/harness-core/45640#-NRREFJYh3vlkIX6EhLv-r2-8:-NRnz9ZE31nFFOT85aEt:bwt0vn6) ([raw file](https://github.com/harness/harness-core/blob/8c3533e9f3062bfca74f0b8830d7d5f6cc91e5b9/930-delegate-tasks/src/main/java/software/wings/helpers/ext/bamboo/BambooCollectionTask.java#L8)):*
<details><summary><i>Previously, gn-harness (Gaurav Nanda) wrote…</i></summary><blockquote>

all these package name changes will lead to Kryo exceptions, right?
</blockquote></details>

I have changed the packages for 5 classes and none of them are in `manager-kryo-registrations.txt` and `kryo-registrations.txt` so ideally they should not cause any Kryo exceptions.

**Comment by Raghav-Murali** [comment]
trigger ut2, ut5

**Review by gn-harness — COMMENTED** [review]
> Reviewed 107 of 154 files at r1, 43 of 48 files at r2, 5 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45640)* status: 5 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained.

**Review by gn-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45640)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Review by mdmilic — APPROVED** [review]
> Reviewed 5 of 5 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45640)* status: 1 file(s) to review. 1 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/software/wings/service/intfc/security/BUILD.bazel` line 25 at r2](https://reviewable.io/reviews/harness/harness-core/45640#-NRjfoiPCltFgw7ORCZK:-NRssiOA0TlXDU8fXLkx:b-wlpj51) ([raw file](https://github.com/harness/harness-core/blob/8c3533e9f3062bfca74f0b8830d7d5f6cc91e5b9/930-delegate-tasks/src/main/java/software/wings/service/intfc/security/BUILD.bazel#L25)):*
<details><summary><i>Previously, Raghav-Murali wrote…</i></summary><blockquote>

That's a good catch. Need to look into the BuildCleaner a bit on why the 960-persistence module got added. There is one dependency on it though -

In `930-delegate-tasks/src/main/java/io/harness/secrets/setupusage/BUILD.bazel` has below dependency -
"//960-persistence/src/main/java/io/harness/persistence/api:module"
</blockquote></details>

I guess the important part for delegate was removing morphia, I checked this one doesn't bring in morphia, so should be ok for now.

**Review by raghuAtWings — APPROVED** [review]
> Reviewed 106 of 154 files at r1, 43 of 48 files at r2, 5 of 5 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45640)* status: :shipit: complete! ✓

**Comment by Raghav-Murali** [comment]
trigger ut1, ut6, ut7

**Comment by Raghav-Murali** [comment]
trigger ut2

**Pull request merged**